source/rules/rxjsSuffixSubjectsRule.ts

```diff
@@ -0,0 +1,146 @@
+import { couldBeType } from "../support/util";
+
+const defaultTypesRegExp = /^EventEmitter$/;
+
+export class Rule extends Lint.Rules.TypedRule {
+  public static metadata: Lint.IRuleMetadata = {
+    description:
+      "Ensures subjects are suffixed with 'Subject'.",
+    options: {
+      properties: {
+        functions: { type: "boolean" },
+        methods: { type: "boolean" },
+        parameters: { type: "boolean" },
+        properties: { type: "boolean" },
+        suffix: { type: "string" },
+        types: { type: "object" },
+        variables: { type: "boolean" }
+      },
+      type: "object"
+    },
+    optionsDescription: Lint.Utils.dedent`
+      An optional object with optional \`functions\`, \`methods\`, \`parameters\`,
+      \`properties\` and \`variables\` properties.
+      The properies are booleans and determine whether or not subjects of that particular expression need a suffix.
+      \`parameters\`, \`properties\` and \`variables\` default to \`true\`.
+      The object also has optional \`types\` properties which are themselves
+      objects containing keys that are regular expressions and values that are booleans -
+      indicating whether suffixing is required for particular types.`,
+    requiresTypeInfo: true,
+    ruleName: '"rxjs-suffix-subjects"',
+    type: "style",
+    typescriptOnly: true
+  };
+
+  private SUFFIX: string;
+  private types: { regExp: RegExp; validate: boolean }[] = [];
+  private validate = {
+    functions: false,
+    methods: false,
+    parameters: true,
+    properties: true,
+    variables: true
+  };
+  public readonly FAILURE_MESSAGE = (identifier: string) =>
+    `Subject '${identifier}' must be suffixed with '${this.SUFFIX}'.`
+
+  public applyWithProgram(
+    sourceFile: ts.SourceFile,
+    program: ts.Program
+  ): Lint.RuleFailure[] {
+    const failures: Lint.RuleFailure[] = [];
+    const typeChecker = program.getTypeChecker();
+
+    // get configurations
+    const { ruleArguments } = this.getOptions();
+    const [options] = ruleArguments;
+    if (options) {
+      this.SUFFIX = options.suffix;
+      if (options.types) {
+        Object.entries(options.types).forEach(
+          ([key, validate]: [string, boolean]) => {
+            this.types.push({ regExp: new RegExp(key), validate });
+          }
+        );
+      } else {
+        this.types.push({ regExp: defaultTypesRegExp, validate: false });
+      }
+      this.validate = { ...this.validate, ...options };
+    }
+
+    let identifiers: ts.Node[] = [];
+
+    if (options.functions) {
+      identifiers = identifiers.concat(
+        tsquery(
+          sourceFile,
+          `:matches(FunctionDeclaration, FunctionSignature) > Identifier`
+        )
+      );
+    }
+
+    if (options.methods) {
+      identifiers = identifiers.concat(
+        tsquery(
+          sourceFile,
+          `:matches(MethodDeclaration, MethodSignature) > Identifier`
+        )
+      );
+    }
```

Comment:
I can see that functions and methods are used. So you can disregard the first comment in this review. However, I think functions and methods should be covered in the tests.

Also, I think that maybe the `functions` and `methods` options should default to `false`. What do you think? Do you use a suffix for functions and methods? Given your other rule, I suspect you don't have any functions or methods that return subjects.